src/linux/common.rs

```diff
@@ -65,3 +67,40 @@ pub fn convert(code: c_uint, state: c_uint, type_: c_int, x: f64, y: f64) -> Opt
         })
     }
 }
+
+pub struct Display {
+    display: *mut xlib::Display,
+}
+
+impl Display {
+    pub fn new() -> Option<Display> {
+        unsafe {
+            let display = xlib::XOpenDisplay(null());
+            if display.is_null() {
+                return None;
+            }
+            Some(Display { display })
+        }
+    }
+
+    pub fn get_size(&self) -> Option<(u64, u64)> {
+        unsafe {
+            let screen_ptr = xlib::XDefaultScreenOfDisplay(self.display);
+            if screen_ptr.is_null() {
+                return None;
+            }
+            let screen = *screen_ptr;
+            Some((
+                screen.width.try_into().ok()?,
+                screen.height.try_into().ok()?,
+            ))
+        }
+    }
+}
+impl Drop for Display {
```

Comment:
This is a re implementation of functionality the rust bindings to XCB already include. see https://rtbo.github.io/rust-xcb/src/xcb/base.rs.html#595-614. The rust bindings to XCB are way easier to use, and have built-in drop implementations, making memory management a breeze.